### PR TITLE
Inline some serialize impls

### PR DESCRIPTION
**Pull request by ecstatic-morse**
Some trivial methods in impls of the `Serialize::Encoder` trait are not marked `#[inline]` but are not generic and are (I believe) used cross-crate.

r? @eddyb

### CODE REVIEW

File: src/librustc_middle/ty/query/on_disk_cache.rs [hunk 1]
```diff
@@ -909,10 +909,12 @@ where
 
 macro_rules! encoder_methods {
     ($($name:ident($ty:ty);)*) => {
-        #[inline]
-        $(fn $name(&mut self, value: $ty) -> Result<(), Self::Error> {
```
Comment (jonas-schievink) [inline src/librustc_middle/ty/query/on_disk_cache.rs:913]: Haha, this one is great!

**Comment by ecstatic-morse** [comment]
Presuming this will ultimately be approved, do we want to just rollup=never this or do a manual perf run now? rollup=never saves one perf run if this does have an impact, and the results from  #69281 indicate that it will probably have a small one.

**Comment by nnethercote** [comment]
I always do a perf run ahead of time for any change that is intended to improve performance, because sometimes things don't work the way you'd expect.

@bors try @rust-timer queue

File: src/librustc_middle/ty/query/on_disk_cache.rs (hunk 1 above)
Comment (eddyb) [reply src/librustc_middle/ty/query/on_disk_cache.rs:913]: Oh dear, I just realized what's going on now.

**Review by eddyb — APPROVED** [review]

**Comment by eddyb** [comment]
r=me after perf run

**Comment by eddyb** [comment]
Oh and might as well: @bors rollup=never

**Comment by ecstatic-morse** [comment]
This caused both instruction count and task-clock measurements to regress slightly. I might try a run with only `emit_unit` inlined, since that is what could be affecting #71044, although obviously I'm less confident now. I guess sometimes things don't work out the way you'd expect.

**Comment by eddyb** [comment]
Looks like a wash to me, tbh. I wonder if none of these actually matter in practice. I recall @nnethercote did some work with encoding/decoding, maybe they found everything hot enough.

**Comment by ecstatic-morse** [comment]
So I think we can close this. I'm gonna try making `emit_unit` inline as part of #71044 to see if that addresses the slowdown during metadata encoding I'm seeing there.

I think we should do something about the dangling `#[inline]` though. Maybe just remove it?

**Comment by Dylan-DPC** [comment]
Closing this based on the above comment

**Pull request closed without merging**